shared/js/background/utils.es6.js

```diff
@@ -91,6 +92,29 @@ function getUpgradeToSecureSupport () {
     return upgradeToSecureSupport
 }
 
+// Chrome errors with 'beacon', but supports 'ping'
+// Firefox only blocks 'beacon' (even though it should support 'ping')
+const beaconNamesByBrowser = {
+    'chrome': 'ping',
+    'moz': 'beacon'
+}
+const beaconName = beaconNamesByBrowser[browser]
+
+// Add beacon or ping to the blocked types of requests
+// Depending on browser
+function updateRequestListenerTypes () {
+    if (!constants.requestListenerTypes ||
+        constants.requestListenerTypes.find(beaconName)) {
+        return
+    }
+
+    constants.requestListenerTypes.push(beaconName)
```

Comment:
Sorry, I don't wanna end up in a "too many reviewers" situation but it seems really odd for `constants` to be edited on-the-fly. It goes against the expectation the name `constants` implies.

What do you think about just swapping this line over with a locally defined array which contains both `constants.requestListenerTypes` and either `ping` or `beacon`? https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/cdc7c2ea72fc9d133a88befd2e2205cf3ce3d5cf/shared/js/background/chrome-events.es6.js#L32

I'm not sure this needs to live in Utils, either.

Reply:
Thanks @andrey-p, this is a really good point. I'm now making a shallow copy in chrome events and adding the beacon/ping type there.
However, I left getBeaconName() in utils since it's used elsewhere and it looks like a good, intuitive place where to find it.
Thanks!